sdk/core/azure-core/src/http/transport_policy.cpp

```diff
@@ -42,6 +42,7 @@ namespace Azure { namespace Core { namespace Http { namespace Policies { namespa
     // Windows, when both are defined.
 #if defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
     return ::AzureSdkGetCustomHttpTransport();
+    transportOptions;
```

Comment:
The way we typically do this for unused parameters is as follows:
```suggestion
    (void)transportOptions;
    return ::AzureSdkGetCustomHttpTransport();
```

For example:
https://github.com/Azure/azure-sdk-for-cpp/blob/562ca4c69c2d2887b3241eefe2aae9a33c4037cf/sdk/storage/azure-storage-blobs/src/blob_client.cpp#L801

Reply:
What's the point of the '(void)' cast? C linters care about this, but this is C++ code, not C code. Also, why does this use a C-style cast? c-style casts are a code smell issue in C++ code (See also [avoid casts](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts)). The C++ guidelines *explicitly* call out casting to (void) as a bad thing (specifically in the context of ignoring [[nodiscard]]).

And why put the variable access in live code (as opposed to putting it in the dead code area behind the return statement)?

Reply:
I don't think the "avoid casts" guideline applies here.
I am fine with any reasonable alternative that we agree upon and apply across the repo, consistently, rather than choosing personal stylistic preference.

The `(void)` cast pattern for unused params is more discoverable/searchable, which is good. I like it (or at least have gotten used to it). Having it after a return (unreachable code) is certainly odd to me and would prefer we didn't do that. Similarly, not having anything but the "empty statement", is a bit strange as well.